### ISSUE TITLE
form builder Repeater when or hidden not work

### ISSUE THREAD

**Issue by febelery** [laravel-filament/filament#996]
### Package

filament/forms

### Package Version

v2.0.0

### Laravel Version

v8.6

### Livewire Version

v2.0

### PHP Version

PHP 8.1

### Bug description

```php

Forms\Components\Repeater::make('field')->schema([
        Forms\Components\Select::make('type')
                                ->options([
                                    'input' => 'input something',
                                    'textarea' => 'text area',
                                ])
                                ->reactive()
                                ->required()
                                ->default('input'),
       Forms\Components\TextInput::make('max')
                                ->when(fn(callable $get) => $get('type') === 'textarea'),
])
```

The filed `max` not show when `type` is `textarea`

**Comment by danharrin** [comment]
Hey - someone had issues with this the other day. Can you make sure you're on the latest version of Filament? Also - please swap that `==` for `===`, that weirdly caused someone else issues as well.

**Comment by febelery** [comment]
Yes, I'm sure I'm on the latest version.

```php
Forms\Components\Repeater::make('field')->schema([
        Forms\Components\Select::make('type')
                                ->options([
                                    'input' => 'input something',
                                    'textarea' => 'text area',
                                ])
                                ->required()
                                ->default('input'),
       Forms\Components\TextInput::make('max')
                                ->when(fn(callable $get) => $get('type') === 'textarea'),
])
```
that's not work, but

```php
Forms\Components\Repeater::make('field')->schema([
        Forms\Components\Select::make('type')
                                ->options([
                                    'input' => 'input something',
                                    'textarea' => 'text area',
                                ])
                                ->required()
                                ->default('input'), // notice
       Forms\Components\TextInput::make('max')
                                ->when(fn(callable $get) => $get('type') === 'input'), // notice
])
```
work on fine.

The difference between them is that when the **default is displayed** then it works fine, and when the **default is not displayed** then it is not responsive！

**Comment by febelery** [comment]
I upgrade Filament version problem solved.
But I found some new problem.

When filed is `TagsInput`, js was an errors
```php
Forms\Components\Repeater::make('field')->schema([
        Forms\Components\Select::make('type')
                                ->options([
                                    'input' => 'input,
                                    'select' => 'select',
                                ])
                                ->reactive(),
       Forms\Components\TagsInput::make('options')
                                ->when(fn(callable $get) => $get('type') === 'select'),
])
```
```js
Livewire Entangle Error: Livewire property 'data.fields.c06e037c-c966-4f79-9dcd-6d38260dc20f.options' cannot be found

Alpine Expression Error: Cannot read properties of undefined (reading 'length')
Expression: "state.length || true"

module.esm.js:1653     
Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at eval (eval at <anonymous> (module.esm.js:1697), <anonymous>:3:38)
```

**Comment by febelery** [comment]
<img width="1409" alt="WX20211226-152237@2x" src="https://user-images.githubusercontent.com/52574911/147401677-289a48ab-45c1-4398-86b0-3a1201981be0.png">

**Comment by martinmildner** [comment]
@febelery - See line 4: 'input' => 'input**'**

**Comment by danharrin** [comment]
Fixed by #1008.
